doc/man7/ossl-guide-quic-client-non-block.pod

```diff
@@ -0,0 +1,440 @@
+=pod
+
+=begin comment
+
+NB: Changes to the source code samples in this file should also be reflected in
+demos/guide/quic-client-non-block.c
+
+=end comment
+
+=head1 NAME
+
+ossl-guide-quic-client-non-block
+- OpenSSL Guide: Writing a simple nonblocking QUIC client
+
+=head1 SIMPLE NONBLOCKING QUIC CLIENT EXAMPLE
+
+This page will build on the example developed on the
+L<ossl-guide-quic-client-block(7)> page which demonstrates how to write a simple
+blocking QUIC client. On this page we will amend that demo code so that it
+supports nonblocking functionality.
+
+The complete source code for this example nonblocking QUIC client is available
+in the B<demos/guide> directory of the OpenSSL source distribution in the file
+B<quic-client-non-block.c>. It is also available online at
+L<https://github.com/openssl/openssl/blob/master/demos/guide/quic-client-non-block.c>.
+
+As we saw in the previous example an OpenSSL QUIC application always uses a
+nonblocking socket. However, despite this, the B<SSL> object still has blocking
+behaviour. When the B<SSL> object has blocking behaviour then this means that
+it waits (blocks) until data is available to read if you attempt to read from
+it when there is no data yet. Similarly it waits when writing if the B<SSL>
+object is currently unable to write at the moment. This can simplify the
+development of code because you do not have to worry about what to do in these
+cases. The execution of the code will simply stop until it is able to continue.
+However in many cases you do not want this behaviour. Rather than stopping and
+waiting your application may need to go and do other tasks whilst the B<SSL>
+object is unable to read/write, for example updating a GUI or performing
+operations on some other connection or stream.
```

Comment:
The way this demo is written, it really doesn't allow that. It will just "block" in the select call instead.

Reply:
I did it this way on purpose for the sake of simplicity. I could have added additional complexity to the demos showing "real" work being done - but in reality every application will want to do this the same way and I did not want to confuse the issue with code which isn't really OpenSSL programming per se. I have however added a big comment in the code explaining where to add your code to do "real" work, and outlining how you might do that.